cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -877,9 +877,9 @@ private Future<MapAndStatus<ConfigMap, KafkaRebalanceStatus>> onPendingProposal(
                                             }
                                         })
                                         .onFailure(e -> {
-                                            LOGGER.errorCr(reconciliation, "Cruise Control getting rebalance proposal failed", e.getCause());
+                                            LOGGER.errorCr(reconciliation, "Cruise Control getting rebalance proposal failed: " + e.getCause());
```

Comment:
e.getCause() gives us the exact error which Cruise Control is throwing and not CruiseControlRestException(our custom exception)which is wrapping the exception thrown by CC.

Reply:
What is the test case you are talking about?

Reply:
Well, normally you do want to get either the stack trace from the exception of just some error message. I'm not sure I understand how useful the current output you suggest it. Maybe you can provide some examples.

Reply:
I have added the example logs in the PR description

Reply:
I'm not sure what does this really add:
```
2023-06-05 09:26:44 ERROR KafkaRebalanceAssemblyOperator:880 - Reconciliation #6(kafkarebalance-watch) KafkaRebalance(myproject/my-rebalance): Cruise Control getting rebalance proposal failed: java.lang.Exception: com.linkedin.kafka.cruisecontrol.exception.OptimizationFailureException
```

The name of the exception class is meaningless. And there are no additional details such as stacktrace to look it up. That is why you normally log the exception as it was originally to have the information needed to make some use out of it.

Reply:
@fvaleri Earlier the ‘e.getCause’ was coming out as null. Now what Iam doing is setting the cause by snipping it out from the json response coming from Cruise Control. Now Iam able to set the cause exactly the same as whats coming from CC. So now iam able to set both the error message as well as the exception cause in the logs. I also was trying to propogate the exception cause to the status also but it was being a bit troublesome. I did made the same change earlier as you are suggestion but then we wouldn’t be able to get the cause since it would have been null and just the error message

Reply:
PS: The end of the condition message looks also a bit weird with `... with brokers.''.'`.

Reply:
Okay, I will fix this

Reply:
@ShubhamRwt my point was that we don't need to create an extra class to fill the exception cause. That error always comes from the CC response, so just use getMessage and maybe improve that weird ending.

Reply:
Okay got it. Thanks @fvaleri